fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/cursors/OrElseCursor.java

```diff
@@ -57,7 +65,44 @@
     public OrElseCursor(@Nonnull RecordCursor<T> inner, @Nonnull Function<Executor, RecordCursor<T>> func) {
         this.inner = inner;
         this.func = func;
-        this.first = true;
+        this.state = State.UNDECIDED;
+    }
+
+    public OrElseCursor(@Nonnull Function<byte[], ? extends RecordCursor<T>> innerFunc,
```

Comment:
This feels like it should be `INTERNAL`, but I could be wrong

Reply:
Sure, why not.